nextflow_schema.json

```
@@ -522,6 +522,59 @@
+                    "description": "Assigns the database /s to be used to annotate the coding regions."
+                },
+                "protein_annotation_interproscan_enableprecalc": {
+                    "type": "boolean",
+                    "help_text": "This increases the speed of functional annotation with InterProScan by pre-calculating matches found in the UniProtKB, thereby identifying unique matches in the query sequences for faster annotation. By default this is turned OFF.\n\nFor more information about this flag see the tool [documentation](https://interproscan-docs.readthedocs.io/en/latest/HowToRun.html).\n\n> Modifies tool parameter(s):\n> - InterProScan: `---diasable-precalc`",
+                    "description": "Pre-calculates residue mutual matches."
+                },
+                "protein_annotation_interproscan_enableresidueannot": {
+                    "type": "boolean",
+                    "help_text": "Activating this flag  will remove the residue annotation from the final XML and JSON output files, which we do not supply as output files from InerProScan. \n\nFor more information about this flag see the tool [documentation](https://interproscan-docs.readthedocs.io/en/latest/HowToRun.html).\n\n> Modifies tool parameter(s):\n> - InterProScan: `--disable-residue-annot`",
+                    "description": "Enable residue annotation in XML and JSON files."
+                },
+                "protein_annotation_interproscan_disableresidueannottsv": {
+                    "type": "boolean",
+                    "help_text": "This disables the addition of the annotations assigned according to the databases activated to the final `<sample>_interproscan.tsv` file.  Turning this on will remove the annotations from the final table. It is not recommended to use this option, as it will cause a run failure because the format of the resulting files will no longer be adequate for integration in the final summary tables. Currently, only applicable for AMPcombi2. \n\nFor more information about this flag see the tool [documentation](https://interproscan-docs.readthedocs.io/en/latest/HowToRun.html).\n\n> Modifies tool parameter(s):\n> - InterProScan: `--enable-tsv-residue-annot`\n",
```

```suggestion
                    "help_text": "This disables the addition of the annotations assigned according to the databases activated to the final `<sample>_interproscan.tsv` file. Turning this on will remove the annotations from the final table. It is not recommended to use this option, as it will cause a run failure because the format of the resulting files will no longer be adequate for integration in the final summary tables. Currently, only applicable for AMPcombi2. \n\nFor more information about this flag see the tool [documentation](https://interproscan-docs.readthedocs.io/en/latest/HowToRun.html).\n\n> Modifies tool parameter(s):\n> - InterProScan: `--enable-tsv-residue-annot`\n",
                    "fa_icon": "fas fa-eraser",
```

Do we need this parameter at all if it breaks the pipeline when changing the default?

---

Isnt it better to include it here and say why that should noot be touched rather than the user adding it later as an `èxt.arg` ?

---

They cannot add it by ext.arg :D Everything that should be ext.arg is a pipeline parameter. If we don't offer it, the user cannot use it. That's why we add as much as necessary and leave out everything that shouldn't be touched (or is too irrelevant).

---

We just discussed on Slack ([here](https://nfcore.slack.com/archives/C043FMKUNLB/p1739198611742959)), that it might be safer to throw a warning in the AMP workflow, if the user changes this interproscan flag. Because we shouldn't expect them to read the whole params page or give them the pipeline-supported option to switch this flag and thereby crash the pipeline :see_no_evil: